pages/settings/profile.tsx

```diff
@@ -1,57 +1,107 @@
-import { Avatar, Button, FlagSection, Layout } from "components";
-import { NextPage } from "next";
-import styles from "pages/settings/Profile.module.scss";
+import { Avatar, Button, FlagSection, Layout, TextInput } from "components";
+import profileStyles from "pages/settings/Profile.module.scss";
+import { NextPageWithLayout } from "../_app";
+import { useAuth } from "hooks/useAuth";
 import { PERSON_FALLBACK_IMAGE_URL } from "utils/constants";
 
-export const ProfilePage: NextPage = () => {
+export const ProfilePage: NextPageWithLayout = () => {
+  const user = useAuth({ redirectTo: "/login?next=settings/profile" });
+  if (!user) return null;
```

Comment:
I really want to move authentication to the next server side so we can avoid doing this! Lemme know if you have any ideas...its kindof tricky because the Next server doesn't have access to the http only cookie that the client is using to authenticate its requests.  Out of scope for this PR but something I'm thinking about

Reply:
This will be of use perhaps - I can try it out https://github.com/maticzav/nookies